### PR TITLE
Logging refactored

### PR DESCRIPTION
**Pull request by babenek**
According customer experience:
- remove top-level logging initialization
- using get_logger to apply external logger which was configured by user for their app
- Add top level exception and return code from __main__ in the case

### CODE REVIEW

**Review by xDizzix — APPROVED** [review]
Thanks for support

File: credsweeper/validations/stripe_api_key_validation.py [hunk 1]
```diff
@@ -30,7 +32,7 @@ def verify(cls, line_data_list: List[LineData]) -> KeyValidationOption:
         try:
             r = requests.get("https://api.stripe.com/v1/charges", auth=(line_data_list[0].value, ""))
         except (requests.exceptions.ConnectionError, Exception) as exc:
-            logging.info(exc)
+            logger.info(exc)
```
[the remark this answers is not in the document]
Comment (babenek) [reply credsweeper/validations/stripe_api_key_validation.py:35]: Thanks for review!
Requests exception actually has various messages what happened: 
- timeout
- server error
- no route to host
- ...
Probably there need to split exception types, but it is out of scope.

File: credsweeper/app.py [hunk 2]
```diff
@@ -225,7 +228,7 @@ def file_scan(self, content_provider: ContentProvider) -> List[Candidate]:
 
         """
         candidates: List[Candidate] = []
-        logging.debug(f"Start scan file: {content_provider.file_path}")
+        logger.debug("Start scan file: %s", content_provider.file_path)
```
[the remark this answers is not in the document]
Comment (babenek) [reply credsweeper/app.py:231]: There was good notify fro @xDizzix  - f"..{}" does string formatting even the log line will be not printed. That old style does not. 
I changed debug only due the lines are not displayed often.

**Review by kmnls — APPROVED** [review]

File: credsweeper/app.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (csh519) [reply credsweeper/app.py:233]: I didn't fully understand about this..
So `f` style formatting has concern about will be not printed?
Could you please explain it more?

[the remark this answers is not in the document]
Comment (babenek) [reply credsweeper/app.py:233]: i mean, __str__ of arguments will not be called if loglevel  is less than configured:
```
    def debug(self, msg, *args, **kwargs):
        """
        Log 'msg % args' with severity 'DEBUG'.

        To pass exception information, use the keyword argument exc_info with
        a true value, e.g.

        logger.debug("Houston, we have a %s", "thorny problem", exc_info=1)
        """
        if self.isEnabledFor(DEBUG):
            self._log(DEBUG, msg, args, **kwargs)
```
<- args just passed.

[the remark this answers is not in the document]
Comment (babenek) [reply credsweeper/app.py:233]: Possibly it is valuable with over 1M iterations. Try this:
```
#!/usr/bin/env python
import logging
import time

logger = logging.getLogger(__name__)

if "__main__" == __name__:

    start_time = time.time()
    n = 1 << 22
    while 0 < n:
        logger.debug(f"Test debug : {n}")
        n -= 1
    print(f"Time Elapsed: {time.time() - start_time}s")

    start_time = time.time()
    n = 1 << 22
    while 0 < n:
        logger.debug("Test debug : %d", n)
        n -= 1
    print(f"Time Elapsed: {time.time() - start_time}s")

```

[the remark this answers is not in the document]
Comment (csh519) [reply credsweeper/app.py:233]: @babenek 

It's strange..
As following your explanation, `Test deubg : ` string should be printed in second try.
But there is no string..

``` bash
(venv) csh519@MacBookPro:~/git/csh519/CredSweeper(main⚡) » python xx.py
Time Elapsed: 1.392773151397705s
Time Elapsed: 1.0505683422088623s
```

[the remark this answers is not in the document]
Comment (babenek) [reply credsweeper/app.py:233]: by default it is disabled. 
4M prints take much more time than 1 sec )
The sample demonstrates slowdown of string format in disabled logging.

[the remark this answers is not in the document]
Comment (csh519) [reply credsweeper/app.py:233]: Ah.. So you modifiend this for speed up?

[the remark this answers is not in the document]
Comment (babenek) [reply credsweeper/app.py:233]: yes, only for the debug which is almost disabled in output

[the remark this answers is not in the document]
Comment (csh519) [reply credsweeper/app.py:233]: Thank you for explanation, I understood now. 💡

**Review by csh519 — APPROVED** [review]
Thank you for refactoring.

LGTM 👍

**Pull request merged**